ado/ipacheckreadreplace.ado

```diff
@@ -1,161 +1,328 @@
-		error 498
-	}
-	cap assert missing(`oldvalue') if missing(`selectmultiple')
-	if _rc {
-		count if !missing(`oldvalue') & missing(`selectmultiple')
-		di as error "oldvalue specified for `r(N)' corrections entries without selectmultiple"
-		error 498
-	}
-	
-	tempfile selectmult allcorrections
-	save `allcorrections'
-
-	keep if !missing(`selectmultiple')
-	count 
-	if `r(N)' > 0 {
-		tostring `value', replace
-		tostring `oldvalue', replace
-		bysort `id': gen j = _n
-		sum j
-		local numloops = `r(max)'
-		levelsof `variable'
-		local correction_vars `r(levels)'
-		reshape wide `variable' `value' `selectmultiple' `oldvalue', i(`id') j(j)
-		save `selectmult'
-		
-		use `allcorrections', clear
-		drop if !missing(`selectmultiple')
-		save `allcorrections', replace
-		
-		restore
-		* merge in the selectmultiples
-		tempvar mult_merge
-		merge 1:1 `id' using `selectmult', gen(`mult_merge')
-		count if `mult_merge' == 2
-		if `r(N)' > 0 {
-			noi di in red "`r(N)' corrections failed to merge"
-			noi list `id' if `mult_merge' == 2
-		}
-		drop if `mult_merge' == 2
-		drop `mult_merge'
-		
-		* replace the selectmultiples
-		tempvar concat
-		foreach var in `correction_vars' {
-			tostring `var', replace
-			split `var', generate(__`var')
-			local stublist `r(varlist)'
-			foreach stub in `stublist' {
-				forval n = 1 / `numloops' {
-					replace `stub' = `value'`n' if `variable'`n' == "`var'" ///
-													& `stub' == `oldvalue'`n'
-				}
-			}
-			egen `concat' = concat(`stublist'), punct(" ")
-			forval n = 1 / `numloops' {
-				replace `var' = `concat' if `variable' == "`var'"
-			}
-			drop `concat' `stublist'
-		}
-		* drop the merged variables
-		forval n = 1 / `numloops' {
-			drop `variable'`n' `value'`n' `selectmultiple'`n' `oldvalue'`n'
-		}
-	}
-	else {
-		restore // if there were no selectmultiple variables
-	}
-	* the selectmultiple phase is complete! Now to add in the other corrections:
-	readreplace using `allcorrections', ///
-			id(`id') variable(`variable') value(`value') use
-	
-	}
-	noi di "Replacements complete."
-	
-	end
-	
-	
-
+*! Vesrion 2.0.0 Ishmail Azindoo Baako, Jan 18
+*! Version 1.0.0 Kelsey Larson, March 16
+
+prog ipacheckreadreplace, rclass
+	/* This program makes corrections to a database from a second database of 
+		corrections. The corrections sheet should have the following columns:
+			 - id: Specifies the uniqueid of the obs to be corrected
+			 - variable: specifies the variable to be corrected
+			 - value: specifies the current value of variable
+			 - newvalue: specifies the new variable for variable
+			 - action: specifies the action to take ie. drop, replace, okay
+		options:
+			 - logusing: correction log file. .xlsx or .xls
+			 - sheet: sheetname
+	*/
+	
+	#d;
+	syntax using/, 
+		id(varname) 
+		VARiable(name) 
+		VALue(name) 
+		NEWVALue(name) 
+		ACTion(name)
+		COMMents(name)
+		[LOGUSing(string) sheet(string)]
+		;
+	#d cr
+	
+	cap version 15.1 

```

Comment:
@iabaako... is this actually the minimum version of Stata required to run? If not we should set it to the minimum. Also we should remove the `cap`.